update firstboot examples in docs and tests

rebmit opened the pull request:
Update the firstboot examples in the documentation and tests to be compatible with systemd v258.

Closes https://github.com/nix-community/preservation/issues/22

rebmit committed:
tests/flake.lock: Update

Flake lock file updates:

• Updated input 'flake-parts':
    'github:hercules-ci/flake-parts/4524271976b625a4a605beefd893f270620fd751?narHash=sha256-%2BuWLQZccFHwqpGqr2Yt5VsW/PbeJVTn9Dk6SHWhNRPw%3D' (2025-09-01)
  → 'github:hercules-ci/flake-parts/52a2caecc898d0b46b2b905f058ccc5081f842da?narHash=sha256-8oNVE8TrD19ulHinjaqONf9QWCKK%2Bw4url56cdStMpM%3D' (2025-11-12)
• Updated input 'flake-parts/nixpkgs-lib':
    'github:nix-community/nixpkgs.lib/a73b9c743612e4244d865a2fdee11865283c04e6?narHash=sha256-x2rJ%2BOvzq0sCMpgfgGaaqgBSwY%2BLST%2BWbZ6TytnT9Rk%3D' (2025-08-10)
  → 'github:nix-community/nixpkgs.lib/719359f4562934ae99f5443f20aa06c2ffff91fc?narHash=sha256-b0yj6kfvO8ApcSE%2BQmA6mUfu8IYG6/uU28OFn4PaC8M%3D' (2025-10-29)
• Updated input 'nixpkgs':
    'github:nixos/nixpkgs/8eb28adfa3dc4de28e792e3bf49fcf9007ca8ac9?narHash=sha256-NOrUtIhTkIIumj1E/Rsv1J37Yi3xGStISEo8tZm3KW4%3D' (2025-09-05)
  → 'github:nixos/nixpkgs/c5ae371f1a6a7fd27823bc500d9390b38c05fa55?narHash=sha256-4PqRErxfe%2B2toFJFgcRKZ0UI9NSIOJa%2B7RXVtBhy4KE%3D' (2025-11-12)

rebmit committed:
docs: update firstboot examples for systemd v258 compatibility

rebmit committed:
docs: update firstboot examples for systemd v258 compatibility

rebmit committed:
tests: fix failed tests using new firstboot-compatible examples

rebmit committed:
tests: fix failed tests using new firstboot-compatible examples

rebmit commented:
Seems like we should also update the workflows.

rebmit committed:
ci: run separate firstboot-bind-mount and firstboot-symlink tests

rebmit committed:
ci: run separate firstboot-bind-mount and firstboot-symlink tests

rebmit commented:
Can confirm the tests build locally, likely hitting https://github.com/NixOS/nixpkgs/issues/399245

rebmit commented:
> Today I had more success with plain config, than with symlink variants mentioned in the docs.

Yes, it **does work**, and for most setups this is sufficient — this setup is also shown in https://nix-community.github.io/preservation/examples.html#simple.

However, in the _first-boot section_, the goal is not only to make machine-id preservation work, but to ensure compatibility with **systemd's first-boot semantics**.

rebmit commented on docs/src/examples.md:
```diff
@@ -69,11 +80,9 @@ mount created by system to the persistent volume.
-  systemd.suppressedSystemUnits = [ "systemd-machine-id-commit.service" ];
+  boot.initrd.systemd.tmpfiles.settings.preservation."/sysroot/persistent/etc/machine-id".f = {
+    argument = "uninitialized";
+  };
```
It might be worth adding an explicit option for the tmpfiles argument, rather than requiring users to write something like `boot.initrd.systemd.tmpfiles.settings.preservation."/sysroot/persistent/etc/machine-id".f.argument`.

That would be more ergonomic and less coupled to the implementation details of preservation, making it less likely to break if the implementation changes in the future.

ThinkChaos replied:
I agree: having to patch the generated tmpfiles rules is awkward.

I see 2 possible improvements (inclusive or'ed):
1. add an option like `initialContent` to `preserveAt.*.files`
2. have an opinionated option to setup machine-id properly: `preservation.machine-id.preserveAt :: path` that sets up everything (a `preserveAt.${config.preservation.machine-id.preserveAt}.files` entry, and patches the tmpfiles rule & `systemd-machine-id-commit.service`)

I think 2 would be nicer in the long term: on top of being more ergonomic than 1 (less specific knowledge needed about machine-id, and hiding implementation details better), it avoids users copy-pasting magic configuration that needs to be updated from time to time, which is easy to miss both initially and when updated.

ThinkChaos reviewed (approved):
The updated documentation works for me, thank you!

rebmit committed:
tests/flake.lock: Update

Flake lock file updates:

• Updated input 'flake-parts':
    'github:hercules-ci/flake-parts/4524271976b625a4a605beefd893f270620fd751?narHash=sha256-%2BuWLQZccFHwqpGqr2Yt5VsW/PbeJVTn9Dk6SHWhNRPw%3D' (2025-09-01)
  → 'github:hercules-ci/flake-parts/f7c1a2d347e4c52d5fb8d10cb4d94b5884e546fb?narHash=sha256-m1Yf0wZ8j1OHjTc2UwHwyQRSnNeSgLJOd7q5Y45hzi4%3D' (2026-05-13)
• Updated input 'flake-parts/nixpkgs-lib':
    'github:nix-community/nixpkgs.lib/a73b9c743612e4244d865a2fdee11865283c04e6?narHash=sha256-x2rJ%2BOvzq0sCMpgfgGaaqgBSwY%2BLST%2BWbZ6TytnT9Rk%3D' (2025-08-10)
  → 'github:nix-community/nixpkgs.lib/f5901329dade4a6ea039af1433fb087bd9c1fe14?narHash=sha256-GOkGPcboWE9BmGCRMLX3worL4EMnsnG8MyKmXNeYuhQ%3D' (2026-04-26)
• Updated input 'nixpkgs':
    'github:nixos/nixpkgs/8eb28adfa3dc4de28e792e3bf49fcf9007ca8ac9?narHash=sha256-NOrUtIhTkIIumj1E/Rsv1J37Yi3xGStISEo8tZm3KW4%3D' (2025-09-05)
  → 'github:nixos/nixpkgs/a799d3e3886da994fa307f817a6bc705ae538eeb?narHash=sha256-3av0pIjlOWQ6rDbNOmpUSvbNnJkGORQKKjb4LtCZsIY%3D' (2026-06-06)

rebmit committed:
tests: unbreak tests after flake update
